model-optimizer/extensions/middle/GatherNdNormalizer.py

```diff
@@ -25,11 +25,14 @@
 from mo.ops.reshape import Reshape
 
 
-class GatherNdNormalize(MiddleReplacementPattern):
+class GatherNDNormalize(MiddleReplacementPattern):
     """
     Hot fix for new speech-to-text model enabling while GatherND is not implemented in IE.
-    We can replace GatherNd to Reshape + Gather in case when GatherNd indices have just one
+    We can replace GatherND to Reshape + Gather in case when GatherND indices have just one
     meaningful dimension.
+    TODO: Investigate whether we must replace GatherND with Reshape + Gather always (due to performance benefits)
```

Comment:
Please, create a separate task for this investigation.

Reply:
Create: 40828